### PR TITLE
OCM-8664 | ci: Support ROSACLI_BUILD ENV variable for the job triggering 

### PR DESCRIPTION
**Pull request by xueli181114**

### CODE REVIEW

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-profile-demo

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-profile-demo

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-private-link-f3

**Comment by xueli181114** [comment]
```
2024-06-11 15:13:20 (190 MB/s) - 'rosa_Linux_x86_64.tar.gz' saved [16275077/16275077]
LICENSE
README.md
rosa
[CI] Overriding the rosa PATH in the image with new PATH /tmp/tmp.NE0TKTSfmv/rosa
[CI] Verify current rosa is using the overrided PATH /tmp/tmp.NE0TKTSfmv/rosa
[CI] rosa is overrided
```

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-profile-demo

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-profile-demo

File: ci-operator/step-registry/rosa/setup/rosa-setup-commands.sh [hunk 1]
```diff
@@ -19,6 +19,45 @@ log(){
...
+  echo "[CI] Verify current rosa is using the overrided PATH $rosaDownloadTempDir/rosa"
+  current_rosa_path=$(which rosa)
+  if [[ "$current_rosa_path" != "$rosaDownloadTempDir/rosa" ]];then
+    echo "[CI] rosa override failed. Current rosa is using $current_rosa_path not from $rosaDownloadTempDir/rosa"
+    exit 1
+  else
+    echo "[CI] rosa is overrided"
+  fi
+ 
+  rosa version
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/setup/rosa-setup-commands.sh:50]: Should be removed or comment out first?

Comment (xueli181114) [reply ci-operator/step-registry/rosa/setup/rosa-setup-commands.sh:50]: It had been fixed.

File: ci-operator/step-registry/rosa/setup/rosa-setup-commands.sh [hunk 2]
```diff
@@ -19,6 +19,45 @@ log(){
...
+  # I didn't use rosa version to verify because there is a known issue will cause nil pointer error for the command
+  # will update to rosa version to verify the overriden one once it is fixed.
+  echo "[CI] Verify current rosa is using the overrided PATH $rosaDownloadTempDir/rosa"
+  current_rosa_path=$(which rosa)
+  if [[ "$current_rosa_path" != "$rosaDownloadTempDir/rosa" ]];then
+    echo "[CI] rosa override failed. Current rosa is using $current_rosa_path not from $rosaDownloadTempDir/rosa"
+    exit 1
+  else
+    echo "[CI] rosa is overrided"
+  fi
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply ci-operator/step-registry/rosa/setup/rosa-setup-commands.sh:48]: bingo. That is my first choice. Updating now.

File: ci-operator/step-registry/rosa/setup/rosa-setup-commands.sh (hunk 1 above)
Comment (xueli181114) [reply ci-operator/step-registry/rosa/setup/rosa-setup-commands.sh:50]: Hmm... let me comment it in case any others using an old version

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-profile-demo

File: ci-operator/step-registry/rosa/tests/rosa-tests-commands.sh [hunk 3]
```diff
@@ -5,7 +5,8 @@ set -o errexit
 set -o pipefail
 
 trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
-
+which rosa
+exit 0
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/tests/rosa-tests-commands.sh:9]: The debug step should be removed.

File: ci-operator/step-registry/rosa/setup/rosa-setup-commands.sh [hunk 4]
```diff
@@ -19,6 +19,45 @@ log(){
     echo -e "\033[1m$(date "+%d-%m-%YT%H:%M:%S") " "${*}"
 }
 
+override_rosa () {
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply ci-operator/step-registry/rosa/setup/rosa-setup-commands.sh]: Updated now.

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-profile-demo

File: ci-operator/step-registry/rosa/tests/rosa-tests-commands.sh
[the remark this answers is not in the document]
Comment (xueli181114) [reply ci-operator/step-registry/rosa/tests/rosa-tests-commands.sh:1]: Fixed. It was missed. Thanks a lot.

**Pull request merged**